packages/editor/src/store/reducer.js

```diff
@@ -349,6 +349,17 @@ export function listViewPanel( state = false, action ) {
 	return state;
 }
 
+/**
+ * This reducer does nothing aside initializing a ref to the list view toggle
+ * We will have a unique ref per "editor" instance.
+ *
+ * @param {Object} state
+ * @return {Object} Reference to the list view toggle button.
+ */
+export function listViewToggleRef( state = { current: null } ) {
+	return state;
```

Comment:
As we add non-serializable data into the store we break the compatibility of Redux Devtools. It's also a mutable reference which to me might have more suitable place for it. We might have to decide whether we want to completely drop support of Redux Devtools as a debugging tool or we could just move this to something simpler like a ref object?

```js
export const listViewToggleRef = React.createRef();
```

Reply:
Where would you put that ref object? We want something unique per editor instance (so per editor store).

Reply:
Actually, there's no need for reducer, we can just use a memoized selector.

Reply:
Couldn't we just expose it as a private api or just define it at the module level for each editor? Why making it a selector if it's not selecting anything from the store?

Reply:
We can't define at the module level because if you have multiple EditorProviders (multiple editor instances), it breaks. What we could do is use another "context provider" but for me this is counter productive, the store is a context provider on its own.

I agree that it's weird but it does the job without any downside.

Reply:
> We can't define at the module level because if you have multiple EditorProviders (multiple editor instances), it breaks.

I think I might have missed something obvious here, but isn't the solution in this PR also suffering from the same issue? The memoized selector is defined at the module level and it will never be invalidated. We can register different editor stores but the selectors still return the same ref object.

```js
wp.data.register(wp.data.createReduxStore('editor-store-1', {...wp.editor.storeConfig}));
wp.data.register(wp.data.createReduxStore('editor-store-2', {...wp.editor.storeConfig}));

unlock(wp.data.select('editor-store-1')).getListViewToggleRef() === unlock(wp.data.select('editor-store-2')).getListViewToggleRef(); // true
```

Reply:
Interesting, so you're saying that two memoized selectors from different store instances will return the same values if they both have the same "dependencies"? I feel like you might have uncovered a bigger bug here :). Shouldn't memoized selectors be specific to the store they're used with?

Reply:
You're passing an empty dependencies array so that basically means never revalidates the cached value. If you pass in `[state]` though then it will revalidate every time any change is made to the store. I don't know the exact problem we're solving here so can't provide further feedback, but I still think we should keep it simple and idiomatic here. I don't think prop drilling is that bad.

Reply:
I disagree to be honest, Prop drilling is bad because it means introducing a bad public API here. I'm going to restore the reducer as a follow-up to prevent the issue you raised. It's not important for this particular reducer to be tracked in redux dev tools. If we figure out an alternative that doesn't involve adding new contexts or new APIs, I'm all for it.

Reply:
I've opened a follow-up here https://github.com/WordPress/gutenberg/pull/57257

And to clarify further. Both the "header" of the editor and the "list view" are going to be refactored as components of the "editor" package to unify both post and site editors. This means if we want to keep prop drilling, we'll have to have a public API in `EditorProvider` or something just to share this prop between the two components (because they're both rendered by the `Layout -> Header` ... And for me that's a no go. This is an internal implementation detail that shouldn't leak to the public API of the editor package.

Reply:
Thanks for the extra context! But I still don't understand the use case here 😅.

Unless we register two different editor stores, we'd still be accessing the same ref object even when we render two different `EditorProvider`. We can achieve this by using the `useSubRegistry` prop, but then the sub registry only exists in that part of the React tree. If the `Header` component can be rendered outside that part of the tree then how would the `Header` know which editor and which `listViewToggleRef` it should connect to? If we want to link the connection between `Header` and `EditorProvider` then there has to be some public API or a higher-level context provider.

> It's not important for this particular reducer to be tracked in redux dev tools.

I'm afraid the default behavior is not as simple as ignoring that part in the debugging tool, sometimes it just crashes the devtools or even the whole page, even when the Redux devtools tab is not active.